### PR TITLE
[Merged by Bors] - chore(Topology/PartialHomeomorph): minor clean-up

### PR DESCRIPTION
**Pull request by grunweg**
- group related results into sections
- move one section up, to a more logical place

-------------

Best reviewed commit by commit; the third commit is purely code motion. (Using the colorMoved option makes this obvious. Unfortunately, that's not available in the Github UI.)

### CODE REVIEW

File: Mathlib/Topology/PartialHomeomorph.lean [hunk 1]
```diff
@@ -199,6 +201,7 @@ protected theorem bijOn : BijOn e e.source e.target :=
 protected theorem surjOn : SurjOn e e.source e.target :=
   e.bijOn.surjOn
 #align local_homeomorph.surj_on PartialHomeomorph.surjOn
+end Basic
```
Comment (sgouezel) [inline Mathlib/Topology/PartialHomeomorph.lean:204]:
```suggestion

end Basic
```
Empty line before the end of the section, for readability. Same thing below.

**Comment by grunweg** [comment]
awaiting-review

File: Mathlib/Topology/PartialHomeomorph.lean [hunk 2]
```diff
@@ -1342,6 +1366,8 @@ theorem trans_transPartialHomeomorph (e : α ≃ₜ β) (e' : β ≃ₜ γ) (f''
 
 end Homeomorph
 
+section OpenEmbedding
+
```
Comment (j-loreaux) [inline Mathlib/Topology/PartialHomeomorph.lean:1370]: wrapping a namespace in a section with no extra bits in between, *especially* when they have the same name is just confusing.
```suggestion
```

File: Mathlib/Topology/PartialHomeomorph.lean [hunk 3]
```diff
@@ -1366,11 +1392,12 @@ lemma toPartialHomeomorph_right_inv {x : β} (hx : x ∈ Set.range f) :
   rwa [toPartialHomeomorph_target]
 
 end OpenEmbedding
+end OpenEmbedding
```
Comment (j-loreaux) [inline Mathlib/Topology/PartialHomeomorph.lean:1395]: same as above
```suggestion
```

File: Mathlib/Topology/PartialHomeomorph.lean [hunk 4]
```diff
@@ -1366,11 +1392,12 @@ lemma toPartialHomeomorph_right_inv {x : β} (hx : x ∈ Set.range f) :
...
 
+/- inclusion of an open set in a topological space -/
+section openInclusion
```
Comment (j-loreaux) [inline Mathlib/Topology/PartialHomeomorph.lean:1398]: same as above
```suggestion
```

File: Mathlib/Topology/PartialHomeomorph.lean [hunk 5]
```diff
@@ -1396,8 +1423,11 @@ theorem partialHomeomorphSubtypeCoe_target : s.partialHomeomorphSubtypeCoe.targe
 #align topological_space.opens.local_homeomorph_subtype_coe_target TopologicalSpace.Opens.partialHomeomorphSubtypeCoe_target
 
 end TopologicalSpace.Opens
+end openInclusion
```
Comment (j-loreaux) [inline Mathlib/Topology/PartialHomeomorph.lean:1426]: same as above
```suggestion
```

File: Mathlib/Topology/PartialHomeomorph.lean [hunk 6]
```diff
@@ -1396,8 +1423,11 @@ theorem partialHomeomorphSubtypeCoe_target : s.partialHomeomorphSubtypeCoe.targe
...
 
 namespace PartialHomeomorph
+/- post-compose with a partial homeomorphism -/
```
Comment (j-loreaux) [inline Mathlib/Topology/PartialHomeomorph.lean:1429]:
```suggestion
namespace PartialHomeomorph

/- post-compose with a partial homeomorphism -/
```

**Review by j-loreaux — COMMENTED** [review]
Thanks! Just minor quibbles.

After fixing the issues below, please wait for CI to finish before merging.

bors d+

File: Mathlib/Topology/PartialHomeomorph.lean (hunk 2 above)
Comment (grunweg) [reply Mathlib/Topology/PartialHomeomorph.lean]: My reasoning was that sections can be collapsed in VS Code, making it easier to see at a glance what's in the file. However, namespaces have the same effect, so I'm convinced. Changed (here and elsewhere).

**Comment by grunweg** [comment]
Green (except for lean4checker, which should pass). Thanks for the review!
bors r+

**Pull request closed without merging**